internal/client/client.go

```diff
@@ -70,17 +90,16 @@ func WrapDebug(c *http.Client, debug bool) *http.Client {
 	if !debug {
 		return c
 	}
-	return &http.Client{
-		Transport: RoundTripFunc(func(req *http.Request) (*http.Response, error) {
-			dumpRequest(req)
-			res, err := c.Transport.RoundTrip(req)
-			if err != nil {
-				return res, err
-			}
-			dumpResponse(res)
-			return res, nil
-		}),
-	}
+	c.Transport = newTransport(c.Transport, map[string]string{}, debug)
+	return c
+}
+
+// WrapHTTPClientWithOauth2 is used to supply oauth2 support to a user provided http.Client.
+func WrapHTTPClientWithOauth2(ctx context.Context, httpClient *http.Client, clientCredentials *clientcredentials.Config) *http.Client {
+	ctx = context.WithValue(oauth2.NoContext, oauth2.HTTPClient, httpClient)
+	client := clientCredentials.Client(ctx)
+	httpClient.Transport = client.Transport
+	return httpClient
 }
```

Comment:
I'd prefer we let the returned client be the one that `oauth2.NewClient()` returns. We may not be able to verify timeouts and such, but it feels cleaner here imo. Do you see any other issues with returning the oauth2 client instead?

This suggestion relates to my comment on `management.New()`, making it accept functional options:

```suggestion
// Client an existing http.Client using the provided OAuth2 configuration.
func Client(ctx context.Context, c *http.Client, cfg *clientcredentials.Config, options ...func(*http.Client)) *http.Client {
	for _, option := range options {
		option(c)
	}
	ctx = context.WithValue(ctx, oauth2.HTTPClient, c)
	return oauth2.NewClient(ctx, cfg.TokenSource(ctx))
}
```